docs/proposal-spawn-on-d.md

```diff
@@ -631,74 +674,84 @@ In SoD we need to glob on:
   - `poll`, `kill`
 - name only, for a given cycle point, to match abstract tasks:
   - `spawn`, `trigger`
+- actually trigger needs both kinds of globbing: we might still want to
+  re-trigger multiple unhandled failed tasks at once (i.e. match in the task
+  pool) for instance.
 
 ### Restart
 
-In SoS we only store gross task state in the DB, and infer prerequisite status
-from that on restart (then rely on dependency negotiation to (re-)satisfy
-prerequisites of waiting tasks at start-up). For SoD we'll need to store actual
-prerequisite status in the DB, not just task state. But that's easy to do.
+In SoS we only store gross task state in the DB, infer prerequisite status
+from that on restart and then rely on dependency negotiation to (re-)satisfy
+prerequisites of waiting tasks at start-up.
+
+In SoD we no longer have dependency negotation so we need to store the full
+prerequisite-satisfaction status of tasks in the DB, not just the gross task
+state. More precisely, task pool snapshots need to list:
+- the prerequisite components (upstream outputs) completed so far
+- the parent_finished status populated so far
 
-Also need to auto-spawn tasks with no prerequisites again, like at cold start.
+### Reload and Restart after graph changes
 
-Similarly for reload, if parentless task defs were added.
+This gets much better under SoD but is still a little difficult. If new task
+definitions are added, they will automatically be spawned on demand so long as
+existing tasks are their parents in the graph. But if they have no parents we
+will still have to insert the first instance at the desired cycle point.
```

Comment:
I didn't put that in because it could still be difficult to choose the actual cycle point, unless perhaps you mean the "first valid cycle point at or beyond the current runahead limit point"? (Note that all tasks are initially spawned into the runahead pool, even if their cycle point is not beyond the limit). For instance, the oldest or newest point in the current pool might seem sensible, but the suite could be very widely spread across cycle points.  However, I'll add a note that we should consider something of that nature...